src/Sarif.Multitool/Rules/SarifValidationSkimmerBase.cs

```diff
@@ -113,31 +144,47 @@ protected virtual void Analyze(ThreadFlowLocation threadFlowLocation, string thr
         {
         }
 
+        protected virtual void Analyze(Tool tool, string toolPointer)
+        {
+        }
+
+        protected virtual void Analyze(VersionControlDetails versionControlDetails, string versionControlDetailsPointer)
+        {
+        }
+
         private void Visit(SarifLog log, string logPointer)
         {
+            Analyze(log, logPointer);
+
             if (log.Runs != null)
             {
                 Run[] runs = log.Runs.ToArray();
                 string runsPointer = logPointer.AtProperty(SarifPropertyName.Runs);
 
                 for (int i = 0; i < runs.Length; ++i)
                 {
-                    Run run = runs[i];
```

Comment:
>run [](start = 24, length = 3)

Here and throughout, I decided to tighten up the code by removing these temporaries and evaluating the expressions at the point of call to the next `Visit` method. I think it's just as clear.